test/hisi_zip_test/test_lib.c

```diff
@@ -41,6 +42,239 @@ void *mmap_alloc(size_t len)
 	return p == MAP_FAILED ? NULL : p;
 }
 
+static void gen_random_data(void *buf, size_t len)
+{
+	int i;
+	uint32_t seed = 0;
+	unsigned short rand_state[3] = {(seed >> 16) & 0xffff, seed & 0xffff,
+					0x330e};
+
+	for (i = 0; i < len >> 3; i++)
+		*((uint64_t *)buf + i) = nrand48(rand_state);
+}
+
+static int calculate_md5(comp_md5_t *md5, const void *buf, size_t len)
+{
+	if (!md5 || !buf || !len)
+		return -EINVAL;
+	MD5_Init(&md5->md5_ctx);
+	MD5_Update(&md5->md5_ctx, buf, len);
+	MD5_Final(md5->md, &md5->md5_ctx);
+	return 0;
+}
+
+static void dump_md5(comp_md5_t *md5)
+{
+	int i;
+
+	for (i = 0; i < MD5_DIGEST_LENGTH - 1; i++)
+		printf("%02x-", md5->md[i]);
+	printf("%02x\n", md5->md[i]);
+}
+
+static int cmp_md5(comp_md5_t *orig, comp_md5_t *final)
+{
+	int i;
+
+	if (!orig || !final)
+		return -EINVAL;
+	for (i = 0; i < MD5_DIGEST_LENGTH; i++) {
+		if (orig->md[i] != final->md[i]) {
+			printf("Original MD5: ");
+			dump_md5(orig);
```

Comment:
seems this is for debug, is it useful to print it out if there is something wrong?